upgrade_analysis/models/upgrade_record.py

```diff
@@ -159,13 +159,17 @@ def get_xml_records(self, module):
         # The order of the keys are important.
         # Load files in the same order as in
         # module/loading.py:load_module_graph
-        paths = []
+        files = []
         for key in ["init_xml", "update_xml", "data"]:
             if not manifest.get(key):
                 continue
             for xml_file in manifest[key]:
                 if not xml_file.lower().endswith(".xml"):
                     continue
                 parts = xml_file.split("/")
-                paths.append(os.path.join(addon_dir, *parts))
-        return paths
+                try:
+                    with open(os.path.join(addon_dir, *parts), "r") as xml_handle:
+                        files.append(xml_handle.read())
+                except UnicodeDecodeError:
```

Comment:
I'm thinking a warning would be good at this point

Reply:
Done! :)